### ISSUE TITLE
[feature]: Ability to remove channel reserve

### ISSUE THREAD

**Issue by niteshbalusu11** [lightningnetwork/lnd#8429]
**Is your feature request related to a problem? Please describe.**
Channel reserve is a major UX problem when working with mobile wallets. It would be very helpful to be able to remove it completely. 

**Describe the solution you'd like**
Right now you can set `remote_chan_reserve_sat` when opening a channel but it can only go as low as the dust limit and I understood recently that 0 values in protobuf means default will be used. We could have a separate conf file setting called `disable-remote-chan-reserve` or a separate Boolean flag in the rpc that can disable the reserve and override `remote_chan_reserve_sat`. 

**Describe alternatives you've considered**
Going as low as the dust limit. But it doesn't solve the issue.

**Comment by ProofOfKeags** [comment]
Can you explain why having a value lower than the dust limit would fix your UX problems? We can't (shouldn't?) create an output that is lower than the dust limit, but maybe you're suggesting that there is no remote output in that scenario.

**Comment by niteshbalusu11** [comment]
I am saying allowing for complete removal of reserve will fix the problem. I don't think we should go below the dust limit. It should be 0 or dust to 20% (the way it is today)

Here are some things allowing for removal of reserve helps with:

https://lists.linuxfoundation.org/pipermail/lightning-dev/2023-October/004136.html

**Comment by saubyk** [comment]
Shouldn't this be addressed at the spec level?

**Comment by niteshbalusu11** [comment]
I don't think so. We are already allowing configuration to bring it down to dust level. And it's only remote channel reserve that we are changing.

**Comment by ProofOfKeags** [comment]
Technically this does have to be addressed at the spec level. There is a protocol level [requirement](https://github.com/lightning/bolts/blob/master/02-peer-protocol.md#requirements) that this is set to a value that exceeds the dust limit, however I do think we can come up with a well-defined way to adjust the spec to accommodate this. @t-bast has already expressed a desire to allow for negotiating a 0sat channel reserve.

**Comment by niteshbalusu11** [comment]
This is a little confusing to me. It makes sense if we say me as the opener doesn't want to have a reserve so the peer I am opening to might have a problem with it. In our case, the opener is saying to the peer, you can have 0 zero reserve. There is no reason for the peer to be unhappy about this. The peer would love it.

**Comment by ProofOfKeags** [comment]
I don't disagree with you, but if the spec prohibits it, we need to fix it at the spec level. We can't just change it in LND without coordinating with other actors.

**Comment by ProofOfKeags** [comment]
I just opened a [PR](https://github.com/lightning/bolts/pull/1133) against the spec to relax this, I expect there to be some debate but I think there's a decent chance this will get some traction. Note also that this is also being discussed as an [extension](https://github.com/lightning/blips/pull/30/files) and may find traction there. I'll follow the discussion and once there's an answer at the spec level we can proceed with an implementation plan in LND.

**Comment by t-bast** [comment]
I have indeed published [this blip](https://github.com/lightning/blips/pull/30) to detail how that can be done, and is similar to how we do it for Phoenix. I believe this is very useful to mobile wallets, and it works well in practice.

**Comment by Roasbeef** [comment]
Modifying this issue to be the tracking issue for: https://github.com/lightning/bolts/pull/1140

**Comment by litbot-9000** [comment]
Closing as duplicate of #5485 which tracks the ability to bypass/remove channel reserve requirements.
